lib/msal-browser/src/broker/msg/resp/BrokerHandshakeResponse.ts

```diff
@@ -35,8 +36,9 @@ export class BrokerHandshakeResponse extends BrokerMessage {
             message.data.messageType === BrokerMessageType.HANDSHAKE_RESPONSE &&
             message.data.version) {
             // TODO, verify version compatibility
-            if (trustedBrokerDomains.indexOf(message.origin) < 0) {
-                // TODO make this a browser Error
+            const matchedDomains = trustedBrokerDomains.filter((domain: string) => StringUtils.matchPattern(domain, message.origin));
```

Comment:
Can we double check and add a test to make sure this won't match urls that contain the search string anywhere other than the domain part? e.g. shouldn't match if the url is in the query string

Reply:
The tests for this I think are done already for the `matchPattern` API. If we need to update them I will take a look, but these tests are going to be part of a separate PR.